### PR TITLE
Add missing path argument types

### PR DESCRIPTION
**Pull request by aks-**
- Adds missing path arguments type for app.put, app.post, app.get.
- documents path type for app.METHOD and app.all
- addresses issue https://github.com/expressjs/expressjs.com/issues/630

### CODE REVIEW

File: _includes/api/en/4x/app-get-method.md [hunk 1]
```diff
@@ -12,4 +12,19 @@ subsequent routes if there's no reason to proceed with the current route.
...
+  res.send('GET request to any of above path');
+});
+
+// path could be regex
+app.get(/^startswithboom/, function (req, res) {
+  res.send('GET request to page that starts with startswithboom');
+});
+
+// path could be array of regex
+app.get([/boom/, /bonk.*ers], function (req, res) {
```
Comment (aredridel) [inline _includes/api/en/4x/app-get-method.md]: Missing final slash here.

File: _includes/api/en/4x/app-post-method.md [hunk 2]
```diff
@@ -13,4 +13,19 @@ the current route.
...
+app.post(['/foo/bar', '/other/bar'], function (req, res) {
+  res.send('POST request to any of above path');
+});
+
+// path could be regex
+app.post(/^startswithboom/, function (req, res) {
+  res.send('POST request to page that starts with startswithboom');
+});
+
+// path could be array of regex
```
Comment (aredridel) [inline _includes/api/en/4x/app-post-method.md:27]: Missing final slash here

File: _includes/api/en/4x/app-put-method.md [hunk 3]
```diff
@@ -13,4 +13,19 @@ the current route.
...
+  res.send('PUT request to any of above path');
+});
+
+// path could be regex
+app.put(/^startswithboom/, function (req, res) {
+  res.send('PUT request to page that starts with startswithboom');
+});
+
+// path could be array of regex
+app.put([/boom/, /bonk.*ers], function (req, res) {
```
Comment (aredridel) [inline _includes/api/en/4x/app-put-method.md]: Missing final slash here

**Comment by aks-** [comment]
@aredridel,  thanks for catching that. Fixed it :)

**Comment by blakeembrey** [comment]
Just checking, but did you test the regexps? I don't think `/^startswithboom/` would ever match, since the path will always start with `/`. The other two probably aren't the recommendations we should put in documentation, only if someone will copy it without realising it matches anywhere in the URL. Is it possible to anchor them?

**Comment by aks-** [comment]
@blakeembrey fixed!

File: _includes/api/en/4x/app-get-method.md [hunk 4]
```diff
@@ -12,4 +12,19 @@ subsequent routes if there's no reason to proceed with the current route.
...
 });
+
+// path could be array of strings
+app.get(['/foo/bar', '/other/bar'], function (req, res) {
+  res.send('GET request to any of above path');
+});
+
+// path could be regex
+app.get(/^\/yellboom(er|s)?$/, function (req, res) {
+  res.send('GET request to page that starts with startswithboom');
```
Comment (blakeembrey) [inline _includes/api/en/4x/app-get-method.md:23]: It doesn't start with `startswithboom` anymore 😄

Comment (aks-) [reply _includes/api/en/4x/app-get-method.md:23]: @blakeembrey,  yes the name would be misleading. Hmm it could be  `startswithboomandendswithboom(er|s)`

Comment (dougwilson) [reply _includes/api/en/4x/app-get-method.md:23]: i.m.o. let's use examples that at least seem plausible to exist in an application. For example, `/^\/user\/(create|[0-9]+)$/` at least looks more like something that could exist and help people.

**Comment by crandmck** [comment]
Hi @aks- Thanks for your PR, but I'm going to close it, since the main issue has been addressed by #701.  Again, I want to thank you for brining this issue (back) to my attention.  

You're certainly welcome to open another PR to add more examples to http://expressjs.com/en/4x/api.html#middleware-callback-function-examples.

**Pull request closed without merging**